servicetalk-http-api/src/main/java/io/servicetalk/http/api/StrategyInfluencerChainBuilder.java

```diff
@@ -40,54 +43,72 @@ public StrategyInfluencerChainBuilder() {
     /**
      * Creates a new instance.
      *
-     * @param influencers {@link List} of {@link HttpExecutionStrategyInfluencer}s.
+     * @param influencers {@link List} of {@link ExecutionStrategyInfluencer}s.
      */
-    private StrategyInfluencerChainBuilder(Deque<HttpExecutionStrategyInfluencer> influencers) {
+    private StrategyInfluencerChainBuilder(Deque<ExecutionStrategyInfluencer<?>> influencers) {
         this.influencers = new LinkedList<>(influencers);
     }
 
+    /**
+     * Adds the passed {@link ExecutionStrategyInfluencer} to the head of this chain.
+     *
+     * @param influencer {@link HttpExecutionStrategyInfluencer} to add.
+     */
+    public void prepend(ExecutionStrategyInfluencer influencer) {
+        influencers.addFirst(requireNonNull(influencer));
+    }
+
     /**
      * Adds the passed {@link HttpExecutionStrategyInfluencer} to the head of this chain.
      *
      * @param influencer {@link HttpExecutionStrategyInfluencer} to add.
      */
     public void prepend(HttpExecutionStrategyInfluencer influencer) {
-        influencers.addFirst(requireNonNull(influencer));
+        prepend((ExecutionStrategyInfluencer) influencer);
     }
 
     /**
      * If the passed {@code mayBeInfluencer} is an {@link HttpExecutionStrategyInfluencer} then add it to the head of
      * this chain.
      *
-     * @param mayBeInfluencer An object which may be an {@link HttpExecutionStrategyInfluencer}.
+     * @param mayBeInfluencer An object which may be an {@link ExecutionStrategyInfluencer}.
      * @return {@code true} if the passed {@code mayBeInfluencer} was added to the chain.
      */
     public boolean prependIfInfluencer(Object mayBeInfluencer) {
-        if (mayBeInfluencer instanceof HttpExecutionStrategyInfluencer) {
-            prepend((HttpExecutionStrategyInfluencer) mayBeInfluencer);
+        if (mayBeInfluencer instanceof ExecutionStrategyInfluencer) {
+            prepend((ExecutionStrategyInfluencer) mayBeInfluencer);
             return true;
         }
         return false;
     }
 
+    /**
+     * Append another {@link ExecutionStrategyInfluencer} to this chain.
+     *
+     * @param next {@link ExecutionStrategyInfluencer} to append.
+     */
+    public void append(ExecutionStrategyInfluencer next) {
```

Comment:
`ExecutionStrategyInfluencer` -> `ExecutionStrategyInfluencer<?>` will avoid some class cast warnings (general comment in this class)

Reply:
since this class is in `http.api` can it be type safe on `ExecutionStrategyInfluencer<HttpExecutionStrategy>`?